### PR TITLE
feat: Add missing HTML tags

### PR DESCRIPTION
**Pull request by halostatue**
- **chore: Switch from .tool-versions to .mise.toml**

  This also modernized the configured versions.

- **feat: add missing and new HTML elements & MathML**

  This update is based on the list of tags in MDN.

  - Added widely available tags `hgroup`, `picture`, and `dialog`.
  - Added the new tag `search`, which is part of baseline 2023.
  - Did NOT add `portal` and `fencedframe`; these are experimental and currently available to Chromium-based browsers.
  - Added MathML tags. These have been kept separate following the example of SVG.
  - Fixed a bug in `Temple.Parser.nonvoide_elements/0` and `Temple.Parser.void_elements/0` which was trying to call `Keyword.keys` on a straight list of values.

### CODE REVIEW

**Comment by halostatue** [comment]
Also worth noting: I did not add the missing `template` and `slot` tags for HTML; I think they should be supported, but I’m not sure how `slot` would be implemented in a way that does not conflict with Temple's `slot` implementation.

**Comment by mhanberg** [comment]
> Also worth noting: I did not add the missing `template` and `slot` tags for HTML; I think they should be supported, but I’m not sure how `slot` would be implemented in a way that does not conflict with Temple's `slot` implementation.

The template tag is already included. 

Slot will work but I believe you'd have to call it as `Temple.slot`,

**Comment by halostatue** [comment]
> > Also worth noting: I did not add the missing `template` and `slot` tags for HTML; I think they should be supported, but I’m not sure how `slot` would be implemented in a way that does not conflict with Temple's `slot` implementation.
> 
> The template tag is already included.
> 
> Slot will work but I believe you'd have to call it as `Temple.slot`,

I think we would probably want to add the HTML `slot` as `_slot` since `Temple.slot` is fairly important in the documentation and to how Temple works. I don't have a need for it right now, but it’s worth considering.

**Comment by mhanberg** [comment]
Sorry let me rephrase, if you want to use a slot html element, you can use `Temple.slot`. For any HTML element, you can prepend `Temple.` to it and it'll work the same.

You can also add aliases for elements in config if you didn't want to do that.

I think I would be more open to renaming the slot keyword to not collide with official HTML elements tho, just need to come up with a better name.

Will just defer for now tho.
